src/main/java/easy/gc_coffee_api/usecase/file/UploadFileUseCase.java

```diff
@@ -0,0 +1,75 @@
+package easy.gc_coffee_api.usecase.file;
+
+import easy.gc_coffee_api.dto.file.UploadFileResponseDto;
+import easy.gc_coffee_api.entity.File;
+import easy.gc_coffee_api.exception.GCException;
+import easy.gc_coffee_api.repository.FileRepository;
+import easy.gc_coffee_api.usecase.file.strategy.StorageStrategy;
+import easy.gc_coffee_api.usecase.file.vo.FileUrl;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.util.UUID;
+
+@Service
+public class UploadFileUseCase {
+    private final FileRepository fileRepository;
+    private final StorageStrategy storageStrategy;
+
+    public UploadFileUseCase(
+            FileRepository fileRepository,
+            @Qualifier("localStorage") StorageStrategy storageStrategy
+    ) {
+        this.fileRepository = fileRepository;
+        this.storageStrategy = storageStrategy;
+    }
+
+    public UploadFileResponseDto execute(MultipartFile file) {
+        if (file.isEmpty()) {
+            throw new IllegalArgumentException("빈 파일은 업로드할 수 없습니다.");
+        }
+
+        try {
+            String originalFilename = file.getOriginalFilename();
+            String extension = getExtension(originalFilename); // 확장자
+
+            if(extension.isEmpty()) {
+                throw new IllegalArgumentException("확장자가 없는 파일로 요청했습니다.");
+            }
+
+            String newFileName = UUID.randomUUID() + "." + extension; // 파일명
+            String mimeType = file.getContentType();
```

Comment:
코드 너무 좋게 잘 자짜주셨는데 이부분이 새로운 파일 명을 만드는 부분인데 여기를 함수로 빼거나 다른 클래스에게 위임하는 편은 어떨까요?
```suggestion
            String newFileName = fileNameCreater.excute(file);
```

Reply:
저는 명시적인 코드흐름 보는걸 선호해서 일부러 안했습니다!

Reply:
무슨 의도로 말씀하신건지는 어제 지훈님 코드리뷰하면서 알 것 같아요